File: src/main/java/tech/zolhungaj/amqapi/servercommands/gameroom/game/GameStarting.java
package tech.zolhungaj.amqapi.servercommands.gameroom.game;

import com.squareup.moshi.Json;
import tech.zolhungaj.amqapi.servercommands.Command;
import tech.zolhungaj.amqapi.servercommands.CommandTypeOld;
import tech.zolhungaj.amqapi.servercommands.objects.QuizIdentifier;
import tech.zolhungaj.amqapi.servercommands.objects.QuizPlayer;

import java.util.List;
import java.util.Map;

public record GameStarting(
        @Json(name = "showSelection")
        int showSelection,
        @Json(name = "players")
        List<QuizPlayer> players,
        @Json(name = "groupSlotMap")
        Map<String, List<Integer>> groupSlotMap,
        @Json(name = "multipleChoice")
        boolean multipleChoiceEnabled,
        @Json(name = "quizDescription")
        QuizIdentifier quizIdentifier,
        @Json(name = "gameMode")
        String gameMode
) implements Command {

    @Override
    public String commandName() {
        return CommandTypeOld.GAME_STARTING.commandName;
    }
}

File: src/main/java/tech/zolhungaj/amqapi/servercommands/gameroom/game/GameStarting.kt
package tech.zolhungaj.amqapi.servercommands.gameroom.game

import com.squareup.moshi.Json
import tech.zolhungaj.amqapi.servercommands.CommandType
import tech.zolhungaj.amqapi.servercommands.objects.QuizIdentifier
import tech.zolhungaj.amqapi.servercommands.objects.QuizPlayer

@JvmRecord
@CommandType("Game Starting")
data class GameStarting(
    @Json(name = "showSelection")
    val showSelection: Int,
    @Json(name = "players")
    val players: List<QuizPlayer>,
    @Json(name = "groupSlotMap")
    val groupSlotMap: Map<String, List<Int>>,
    @Json(name = "multipleChoice")
    val multipleChoiceEnabled: Boolean,
    @Json(name = "quizDescription")
    val quizIdentifier: QuizIdentifier,
    @Json(name = "gameMode")
    val gameMode: String
) 